dali/python/nvidia/dali/_multiproc/messages.py

```diff
@@ -13,29 +13,152 @@
 # limitations under the License.
 
 
-class ScheduledTasks:
+import struct
+
+
+class Structure:
+
+    """
+    Utility around Python `struct` module that allows to access and modify `_fields` like an ordinary object attributes,
+    but also read and write their values from/into the buffer in C struct like format.
+    """
+
+    _fields = tuple()
+
+    def __init__(self, *values):
+        self.setup_struct()
+        self.set_values(*values)
+
+    @classmethod
+    def setup_struct(cls):
+        if '_struct_desc' not in cls.__dict__:
+            cls._struct_desc = "@" + "".join(field_type for _, field_type in cls._fields)
+            cls._struct = struct.Struct(cls._struct_desc)
+
+    def __getstate__(self):
+        return self.__dict__.copy()
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self.setup_struct()
+
+    def set_values(self, *values):
+        for (field_name, _), value in zip(self._fields, values):
+            setattr(self, field_name, value)
+
+    def get_values(self):
+        return tuple(getattr(self, field_name) for field_name, _ in self._fields)
+
+    def pack_into(self, buf, offset):
+        return self._struct.pack_into(buf, offset, *self.get_values())
+
+    def unpack_from(self, buf, offset):
+        values = self._struct.unpack_from(buf, offset)
+        self.set_values(*values)
+        return self
+
+    def get_size(self):
+        return self._struct.size
+
+
+class ShmMessage(Structure):
+    _fields = ("worker_id", "i"), ("shm_chunk_id", "i"), ("shm_capacity", "i"), ("offset", "i"), ("num_bytes", "i")
```

Comment:
Do you use the second element of each tuple anywhere? I see it is always referred as `_` and discarded.